crates/sui/src/unit_tests/snapshots/sui__upgrade_compatibility__upgrade_compatibility_tests__all.snap

```diff
@@ -0,0 +1,174 @@
+   ┌─ /fixtures/upgrade_errors/all_v2/sources/UpgradeErrors.move:11:19
+   │
+11 │     public struct StructAbilityMismatchAdd has copy {} // added the copy ability where none existed
+   │                   ^^^^^^^^^^^^^^^^^^^^^^^^ Unexpected ability: 'copy'
+   │
+   = Structs are part of a module's public interface and cannot be changed during an upgrade.
+   = Restore the original abilities of the struct: none for struct 'StructAbilityMismatchAdd'.
+
+error[Compatibility E01003]: ability mismatch
+   ┌─ /fixtures/upgrade_errors/all_v2/sources/UpgradeErrors.move:14:19
+   │
+14 │     public struct StructAbilityMismatchRemove {} // removed the copy ability
+   │                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Missing ability: 'copy'
+   │
+   = Structs are part of a module's public interface and cannot be changed during an upgrade.
+   = Restore the original ability of the struct: 'copy' for struct 'StructAbilityMismatchRemove'.
+
+error[Compatibility E01003]: ability mismatch
+   ┌─ /fixtures/upgrade_errors/all_v2/sources/UpgradeErrors.move:17:19
+   │
+17 │     public struct StructAbilityMismatchChange has drop {} // changed from drop to copy
+   │                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Mismatched abilities: missing 'copy', unexpected 'drop'
+   │
+   = Structs are part of a module's public interface and cannot be changed during an upgrade.
+   = Restore the original ability of the struct: 'copy' for struct 'StructAbilityMismatchChange'.
+
+error[Compatibility E01005]: type parameter mismatch
+   ┌─ /fixtures/upgrade_errors/all_v2/sources/UpgradeErrors.move:20:19
+   │
+20 │     public struct StructTypeParamMismatch<T> { a: T } // changed S to T
+   │                   ^^^^^^^^^^^^^^^^^^^^^^^ Incorrect number of type parameters: expected 2, found 1
+   │
+   = Structs are part of a module's public interface and cannot be changed during an upgrade.
+   = Restore the original struct's type parameters for struct 'StructTypeParamMismatch' including the ordering.
+
+error[Compatibility E01002]: type mismatch
+   ┌─ /fixtures/upgrade_errors/all_v2/sources/UpgradeErrors.move:23:19
+   │
+23 │     public struct StructFieldMismatchAdd {
+   │                   ^^^^^^^^^^^^^^^^^^^^^^ Incorrect number of fields: expected 2, found 3
+   │
+   = Structs are part of a module's public interface and cannot be changed during an upgrade.
+   = Restore the original struct's fields for struct 'StructFieldMismatchAdd' including the ordering.
+
+error[Compatibility E01002]: type mismatch
+   ┌─ /fixtures/upgrade_errors/all_v2/sources/UpgradeErrors.move:30:19
+   │
+30 │     public struct StructFieldMismatchRemove {
+   │                   ^^^^^^^^^^^^^^^^^^^^^^^^^ Incorrect number of fields: expected 2, found 1
+   │
+   = Structs are part of a module's public interface and cannot be changed during an upgrade.
+   = Restore the original struct's fields for struct 'StructFieldMismatchRemove' including the ordering.
```

Comment:
We talked about printing out the full list of field names and types here, but that's complicated because of type parameters not having names in the bytecode, but let's follow up on this.

Reply:
(another follow-up on this vein is handling printing of edge case types, like structs/enum variants with no fields, or positional/anonymous fields).

Reply:
created a ticket